src/multi/content_identifier.cpp

```diff
@@ -0,0 +1,41 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <libp2p/multi/content_identifier.hpp>
+
+#include <libp2p/common/hexutil.hpp>
+
+namespace libp2p::multi {
+
+  ContentIdentifier::ContentIdentifier(Version version,
+                                       MulticodecType::Code content_type,
+                                       Multihash content_address)
+      : version{version},
+        content_type{content_type},
+        content_address{std::move(content_address)} {}
+
+  std::string ContentIdentifier::toPrettyString(const std::string &base) {
+    /// TODO(Harrm): hash type is a subset of multicodec type, better move them
+    /// to one place
+    std::string hash_type = MulticodecType::getName(
+        static_cast<MulticodecType::Code>(content_address.getType()));
+    std::string hash_hex = common::hex_lower(content_address.getHash());
+    std::string hash_length =
+        std::to_string(content_address.getHash().size() * 8);
+    std::string v = "cidv" + std::to_string(static_cast<uint64_t>(version));
+    return base + " - " + v + " - " + MulticodecType::getName(content_type)
```

Comment:
try to use [boost::format](https://www.boost.org/doc/libs/1_66_0/libs/format/doc/format.html)